pkg/kubectl/cmd/config/config.go

```diff
@@ -429,6 +430,10 @@ func getConfigFromFileOrDie(filename string) *clientcmdapi.Config {
 		return clientcmdapi.NewConfig()
 	}
 
+	if config.Clusters == nil {
+		config.Clusters = map[string]*clientcmdapi.Cluster{}
```

Comment:
> I'm wondering if maybe this should be fixed in LoadFromFile instead?
> 
> also, there are several maps in Config which could presumably be set nil by LoadFromFile. Should we cover all of them? should we expect callers to check for nil-ness?

For this structure, I don't see any value in making callers distinguish between nil and empty.  I'd make this change in `LoadFromFile` for all maps there.